src/org/dux/stracetool/StraceCall.java

```diff
@@ -9,20 +9,77 @@
     public final String call;
     public final String[] args;
     public final boolean knownReturn;
-    public final int returnValue;
+    public final int returnValue;        // for strace
+    public final String returnMessage;   // for procmon
 
-    public StraceCall(String call, String[] args, int returnValue) {
+    /**
+     * Factory method to return a new StraceCall object populated from the given
+     * system call, arguments, and return value. Typical usage is for a line of
+     * output from strace (on Linux) for a system call with a return value.
+     * @param call The system call.
+     * @param args The arguments to the system call.
+     * @param returnValue The return value of the system call.
+     * @return A new StraceCall object constructed from the given parameters.
+     */
+    public static StraceCall newLinuxInstanceWithReturn(String call, String[] args, int returnValue) {
+        return new StraceCall(call, args, returnValue);
+    }
+
+    /**
+     * Same as {@code StraceCall.newLinuxInstanceWithReturn} except used for
+     * strace calls that do not have a return value.
+     *
+     * Factory method to return a new StraceCall object populated from the given
+     * system call and arguments. Typical usage is for a line of output from
+     * strace (on Linux) for a system call with no return value.
+     * @param call The system call.
+     * @param args The arguments to the system call.
+     * @return A new StraceCall object constructed from the given parameters.
+     */
+    public static StraceCall newLinuxInstanceNoReturn(String call, String args[]) {
+        return new StraceCall(call, args);
+    }
+
+    /**
+     * Factory method to return a new StraceCall object populated from the given
+     * system call, arguments, and return message. Typical usage is for a line
+     * of output from Process Monitor (on Windows).
+     * @param call The system call.
+     * @param args The arguments to the system call.
+     * @param returnMessage The return message of the system call.
+     * @return A new StraceCall object constructed from the given parameters.
+     */
+    public static StraceCall newWindowsInstance(String call, String[] args, String returnMessage) {
+        return new StraceCall(call, args, returnMessage);
+    }
+
+    private StraceCall(String call, String[] args, int returnValue) {
         this.call = call;
         this.args = Arrays.<String>copyOf(args, args.length);
         this.knownReturn = true;
         this.returnValue = returnValue;
+        this.returnMessage = null;
     }
 
-    public StraceCall(String call, String[] args) {
+    private StraceCall(String call, String[] args) {
         this.call = call;
         this.args = Arrays.<String>copyOf(args, args.length);
         this.knownReturn = false;
         this.returnValue = 0;
+        this.returnMessage = null;
+    }
+
+    private StraceCall(String call, String[] args, String returnMessage) {
```

Comment:
This one is actually a little strange, for the following reason:

You made the StraceCall object immutable with public fields, which is not a bad model. It is very easy to just directly access the fields without having to go through a getter. The problem is that if I have one constructor call into another, it will assign a final field twice. So to make this work, I would need to change the field from being final. 

As soon as I do that, it no longer is safe to be letting clients directly reach in to access the fields, so  I would also need to make them private, and add getter methods. This still wouldn't feel right because it was a good idea to make StraceCall immutable, but now we are not using 'final' to enforce this.

Also, how should we return the args[] array at this point? There could be issues of representation exposure, so we can't just return the whole array. And perhaps copying is unnecessary overhead. So I am currently just returning individual Strings inside of that array, which feels very bad and not extensible at all.

This just seems like it introduces a lot of ugliness to an otherwise good design just to save two lines of code in each constructor (although maybe I am just missing a very simple and clean way to do it?). 

I tried it out anyways. You can check it out [here](https://github.com/omag01/dux/commit/c226b92be32de9d372f35cd3bdb2b98f2d5d729f)
and see if you think it is worth it. If so, I'm happy to make a PR for that as well.